src/client/Microsoft.Identity.Client/Internal/Requests/MsiAuthRequest.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.ApiConfig.Parameters;
+using Microsoft.Identity.Client.Cache.Items;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.PlatformsCommon.Interfaces;
+
+namespace Microsoft.Identity.Client.Internal.Requests
+{
+    internal abstract class MsiAuthRequest : RequestBase
+    {
+        protected readonly AcquireTokenForManagedIdentityParameters _managedIdentityParameters;
+        protected static readonly SemaphoreSlim s_semaphoreSlim = new(1, 1);
+
+        protected MsiAuthRequest(
+            IServiceBundle serviceBundle,
+            AuthenticationRequestParameters authenticationRequestParameters,
+            AcquireTokenForManagedIdentityParameters managedIdentityParameters)
+            : base(serviceBundle, authenticationRequestParameters, managedIdentityParameters)
+        {
+            _managedIdentityParameters = managedIdentityParameters;
+        }
+
+        protected override async Task<AuthenticationResult> ExecuteAsync(CancellationToken cancellationToken)
+        {
+            if (AuthenticationRequestParameters.Scope == null || AuthenticationRequestParameters.Scope.Count == 0)
+            {
+                throw new MsalClientException(
+                    MsalError.ScopesRequired,
+                    MsalErrorMessage.ScopesRequired);
+            }
+
+            ILoggerAdapter logger = AuthenticationRequestParameters.RequestContext.Logger;
+            AuthenticationResult authResult = null;
+
+            IKeyMaterialManager keyMaterial = ServiceBundle.Config.KeyMaterialManagerForTest ??
```

Comment:
The platform proxy should decide what object to give you - actual implementation, null object or test object. Have a look at AbstractPlatformProxy to see how you can get test objects from there.